src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.Analysis.cs

```diff
@@ -342,6 +342,23 @@ internal void ComputeLambdaScopesAndFrameCaptures()
                         }
                     }
                 }
+
+                // Any scope captured by a scope that cannot be a struct
+                // also cannot be a struct
+                var moreNoStructScopes = ArrayBuilder<BoundNode>.GetInstance();
+                foreach (var scope in ScopesThatCantBeStructs)
+                {
+                    var curScope = scope;
+                    while (NeedsParentFrame.Contains(curScope))
+                    {
+                        if (!ScopeParent.TryGetValue(curScope, out curScope))
+                        {
+                            break;
+                        }
+                        moreNoStructScopes.Add(curScope);
```

Comment:
>moreNoStructScopes.Add(curScope); [](start = 24, length = 33)

It feels like we can jump out of this loop as soon as curScope is already in ScopesThatCantBeStructs or in moreNoStructScopes. It might make sense to change moreNoStructScopes to be a HashSet.